homeassistant/components/proxmox/switch.py

```diff
@@ -0,0 +1,77 @@
+"""Support for switches to turn on/off Proxmox VE VMs/Containers."""
+import homeassistant.components.proxmox as proxmox
+from homeassistant.components.switch import SwitchDevice
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up Proxmox VE switch."""
+    nodes = hass.data[proxmox.DATA_PROXMOX_NODES]
+    control = hass.data[proxmox.DATA_PROXMOX_CONTROL]
+    sensors = []
+
+    for node_name in nodes.keys():
+        item = nodes[node_name]
+        if 'type' not in item or item['type'] != 'node':
+            sensor = PXMXSwitch(
+                hass, node_name, 'Switch', control['start'], control['stop'])
+            sensors.append(sensor)
+
+    async_add_entities(sensors)
+
+
+class PXMXSwitch(SwitchDevice):
+    """Switch to turn on / turn off a Proxmox VE Virtual Machine/Container."""
+
+    def __init__(self, hass, node_name, sensor_name, start, stop):
+        """Initialize monitor sensor."""
+        self._hass = hass
+        self._node_name = node_name
+        self._sensor_name = sensor_name
+        self._start = start
+        self._stop = stop
+        self._state = None
+        self._is_available = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} ({})'.format(self._sensor_name, self._node_name)
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if Monitor is available."""
+        return self._is_available
+
```

Comment:
I have the following suggestions:
- add a function for `unique_id`
- add the vmid in the `device_state_attributes`, because if the user changes the name of the entity, he can't directly see the vmid anymore